src/goto-symex/show_program.cpp

```diff
@@ -162,6 +162,7 @@ std::string json_get_key_byte_op_stats(const byte_op_type type)
   case byte_op_type::BYTE_UPDATE:
     return "byteUpdateStats";
   }
```

Comment:
Thought this was legal for switch case over enums! But throws "control reaches end of non-void function" error.

Reply:
This is actually fine! What was probably happening here was that you didn’t have a `default` case. This is not actually an error either, just a warning that if there are any cases you haven’t handled, then you might not reach a return statement which is invalid in a non-`void` function.

This can unfortunately happen even if you explicitly cover all possible cases for the enum, for example:

```
enum class ButtonState { Off = 0, On = 1 };
```

you’d think that only `On` and `Off` are valid values here... unfortunately that’s not how C++ works, so for example in C++11 you can write `ButtonState someInvalidState = static_cast<ButtonState>(10);` – this is somewhat awkward, but is completely intended for use as a way to define new integral types without implicit conversions to existing ones, like for example C++17’s `std::byte`.

In any case, the switch statement isn’t the actual problem here!

Reply:
Thank you! 
Do you have any comments regarding the code in general?